File: 1480.Running_Sum_of_1d_Array/solution.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var runningSum = function (nums) {
  let myArray = [];
  for (let i = 0; i < nums.length; i++) {
    let element = nums[i];
    let x = i - 1;
    while (x >= 0) {
      element += nums[x]
      --x;
    }
    myArray.push(element);
  }
  return myArray;
};

// case 1
console.log(runningSum([1, 2, 3, 4]));

// case 2
console.log(runningSum([1, 1, 1, 1, 1]));

// case 3
console.log(runningSum([3, 1, 2, 10, 1]));

File: 1480.Running_Sum_of_1d_Array/solution.py
def runningSum(nums):
    answer = []
    index = 0
    for num in nums:
        for num2 in nums[:index]:
            num += num2
        answer.append(num)
        index += 1
    else:
        return answer


# case 1
print(runningSum([1, 2, 3, 4]))

# case 2
print(runningSum([1, 1, 1, 1, 1]))

# case 3
print(runningSum([3, 1, 2, 10, 1]))
